docs/bdd-introduction.md

```diff
@@ -0,0 +1,16 @@
+
+## Behavior-Driven Development(BDD):
+
+   Behavior-Driven development (or BDD) is a software development technique that encourages collaboration between developers, QA and non-technical or business participants in a software project.
+
+   BDD focuses on obtaining a clear understanding of desired software behavior through discussion with stakeholders. It extends [TDD](docs/tdd-introduction) by writing test cases in a natural language that non-programmers can read. Behavior-driven developers use their native language in combination with the ubiquitous language of Domain-Driven design to describe the purpose and benefit of their code. This allow the developers to focus on why the code should be created, rather than the technical details, and minimizes translation between the technical language in which the code is written and the domain language spoken by the business, users, stakeholders, project management, etc.
+
+   **"Behavior-driven development is about implementing an application by describing its behavior from the perspective of its stakeholders”** -- Dan North
+   
+    The main intention of a BDD framework is to avoid communication gaps between team members, foster better understanding of the customer and promote continuous communication with real world examples.
+    
+    Let me try to explain with an example, A client wanted to increase his business by developing an important feature to his software.The Client gathered all his stake holders to the meeting where they have discussions about the feature.Mostly the outcome of the discussion will be like [a blind man expalining an elephant](https://en.wikipedia.org/wiki/Blind_men_and_an_elephant). It's inevitable, however the Client reaches to one development team and explains what he really wanted. Sometimes these discussions take shape of user Stroies(in Agile world), these is the time where the developer will handover his hat to Client to describe about feature.After a succesfull description developer has his hat back on implementation.
```

Comment:
I'd like to suggest removing the last 2 paragraphs (from "Let me try to explain...") entirely. Imho the first paragraph is clear enough.